src/main/java/org/example/bookstore/service/book/BookServiceImpl.java

```diff
@@ -81,4 +80,13 @@ public void deleteById(Long id) {
                         "Book not found with id: " + id));
         bookRepository.deleteById(id);
     }
+
+    @Override
+    public List<BookResponseDto> getBookDtoByCategoryId(Long id, Pageable pageable) {
+        return bookRepository
+                .findByCategoriesId(id)
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
```

Comment:
I think you forgot to use it
```suggestion
    @Override
    public List<BookResponseDto> getBookDtoByCategoryId(Long id, Pageable pageable) {
        return bookRepository
                .findByCategoriesId(id, pageable)
                .stream()
                .map(bookMapper::toDto)
                .toList();
    }
```